src/Rules/Between.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Spotwilliams\Validator\Rules;
+
+use Spotwilliams\Validator\Exceptions\ValidationFailed;
+use Spotwilliams\Validator\Contracts\ValidationRule;
+
+class Between implements ValidationRule
+{
+    /** @inheritDoc */
+    public function apply(string $ruleName, string $field, mixed $value = null, $constraint = []): bool
+    {
+        if(!empty($constraint) && is_array($constraint)) {
+            if(!array_key_exists("min", $constraint) || !array_key_exists("max", $constraint)) {
+                throw new ValidationFailed("The field `{$field}` validation requires a min & max constraint.");
+            }
+
+            if (!is_numeric($constraint["min"]) || !is_numeric($constraint["max"])) {
+                throw new ValidationFailed("The validation min & max required numeric value, `{$constraint["min"]}` & `{$constraint["max"]}` given.");
+            }
+
+            if($value < intval($constraint["min"])) {
+                throw new ValidationFailed("The field `{$field}` value is too low, allowed range is between {$constraint["min"]}-{$constraint["max"]}.");
+            }
+
+            if($value > intval($constraint["max"])) {
+                throw new ValidationFailed("The field `{$field}` value is too high, allowed range is between {$constraint["min"]}-{$constraint["max"]}.");
+            }
+
+            return true;
+        } else {
+            throw new ValidationFailed("The field `{$field}` validation requires a min & max constraint.");
+        }
+    }
+}
```

Comment:
I would put all constraint's validations in a separated private method. Will help to improve readability and isolate responsabilities.

Reply:
thank you for your suggestion, ill make improvements according to the advice you provided.

Reply:
hi @spotwilliams  how do you think if I create a new property in the Between class, for example private $field & private $constraint, to make it easier for me to access that property from other methods?